internal/resolve/catalog_test.go

```diff
@@ -462,19 +462,24 @@ func TestCatalogWalker(t *testing.T) {
 }
 
 func buildFooClusterExtension(pkg, channel, version string, upgradeConstraintPolicy ocv1alpha1.UpgradeConstraintPolicy) *ocv1alpha1.ClusterExtension {
-	return &ocv1alpha1.ClusterExtension{
+	ce := &ocv1alpha1.ClusterExtension{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: pkg,
 		},
 		Spec: ocv1alpha1.ClusterExtensionSpec{
 			InstallNamespace:        "default",
 			ServiceAccount:          ocv1alpha1.ServiceAccountReference{Name: "default"},
 			PackageName:             pkg,
-			Channel:                 channel,
 			Version:                 version,
 			UpgradeConstraintPolicy: upgradeConstraintPolicy,
 		},
 	}
+
+	if channel != "" {
+		ce.Spec.Channels = []string{channel}
+	}
```

Comment:
This feels like it should be done via a closure above.

Reply:
Do you mean instead of doing this, do something like:

```go
channels := []string{}
if channel != "" {
	channels = append(channels, channel)
}

return ...
```
?

Reply:
I'm fine with making a change here if you feel strongly about how it is implemented

Reply:
Updated in https://github.com/operator-framework/operator-controller/pull/1173/commits/e8ea4ff20524e3ef9bbdb3a64aeb87221c68a3d7 . Let me know if this is insufficient

Reply:
I was thinking something like this, so the assignment is all in one place. Not a big deal (don't even know if the following is legal Go!)
```
		Spec: ocv1alpha1.ClusterExtensionSpec{
			InstallNamespace:        "default",
			ServiceAccount:          ocv1alpha1.ServiceAccountReference{Name: "default"},
			PackageName:             pkg,
			Channel: func() []string{} {
				if channels != "" {
					return []string{channels}
				}
				return []string{}
			}(),
			Version:                 version,
			UpgradeConstraintPolicy: upgradeConstraintPolicy,
		},

Reply:
I do think that is legal Go, but i'm not sure it is idiomatic. Going to keep it as is for now.